packages/core/src/wot-impl.ts

```diff
@@ -86,7 +109,10 @@ export default class WoTImpl {
 
                 // FIXME should be constrained version that omits instance-specific parts (but keeps "id")
                 let template = td;
```

Comment:
could you please deepclone the TD object? I think it is best practice not to mess around the inputs

Reply:
@fillobotto check this.